### PR TITLE
Return promise for decorator on async method

### PR DESCRIPTION
**Pull request by rickh18**
After updating from 5.3.0 my application broke on a method that was returned a promise. It turned out that the instead of the promise, undefined was returned by the method, comparable to the behaviour described in issue  #106.

This has been fixed by adding `return fn`  in `trackEventMethodDecorator`.
The unit test now has an extra check to see if the original method still works.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "react-tracking",
-  "version": "7.2.1",
+  "version": "7.2.2",
```
Comment (tizmagik) [inline package.json]: Please revert this change, version bumps are part of the release process

File: src/trackEventMethodDecorator.js [hunk 2]
```diff
@@ -22,10 +22,15 @@ export default function trackEventMethodDecorator(trackingData = {}) {
...
         if (Promise && Promise.resolve(fn) === fn) {
-          return fn.then(trackEvent.bind(this)).catch(error => {
-            trackEvent(null, error);
-            throw error;
-          });
+          return fn
+            .then(trackEvent.bind(this))
+            .then(() => {
+              return fn;
+            })
```
Comment (tizmagik) [inline src/trackEventMethodDecorator.js]: nit: mind rewriting as:

```js
.then(() => fn)
```

File: package.json (hunk 1 above)
Comment (rickh18) [reply package.json]: My bad, fixed with 9c47128

File: src/trackEventMethodDecorator.js (hunk 2 above)
Comment (rickh18) [reply src/trackEventMethodDecorator.js]: Fixed with d7adf1f
I also added an e2e test as suggested. 

When writing the test, I noticed that the tracking function breaks if a promise is rejected. It will fail on  `{ value }` since value is set to null after reject. So when using the example in the readme:

` @track((props, state, methodArgs, [{ value }, err]) => { }` <-- will give TypeError

`           .catch(error => {
              trackEvent(null, error);
              throw error;
            });` <-- trackEventMethodDecorators.js line 28, value is set to null

So the happy flow is working now, but maybe this should be fixed as well. (changing null to {}?)

Comment (tizmagik) [reply src/trackEventMethodDecorator.js]: Yea that makes sense, good catch! Mind making that change as a separate PR (will need an accompanying test update)

**Comment by tizmagik** [comment]
Released as [v7.3.0](https://github.com/nytimes/react-tracking/releases/tag/v7.3.0) -- thank you! 🎉
